### ISSUE TITLE
Error loading config file:C:\Users\user\.docker\config.json - open 

### ISSUE THREAD

**Issue by whitecolor** [docker/for-win#867]
When trying to push an image to gcloud with `gcloud --project xxx docker -- push ...` I got the error:
```
WARNING: Error loading config file:C:\Users\user\.docker\config.json - open C:\Users\user\.docker\config.json: The process cannot access the file because it is being used by another process.
```

Have to delete file `.docker\config.json` after it worked fine.

The content of this file was:
```json
{
	"auths": {
		"launcher.gcr.io": {}
	},
	"credsStore": "wincred"
}
```

I'm using latest stable:
Version 17.06.0-ce-win18 (12627)
Channel: stable
d9b6651

I seemed to me that in changelog I've read that something related to this `config.json` file was fixed, but it probably it is not.

**Comment by jasonbivins** [comment]
Hi @whitecolor   The config.json file gets used when we touch a registry or use the Docker Login command - I can recreate this error in Powershell by editing the file directly with Docker running, but not in CMD.  Can you let me know if you see the same thing?

Here's an article explaining how the config.json file gets used.
https://docs.docker.com/engine/reference/commandline/login/

**Comment by whitecolor** [comment]
Probably this file was modified by `gcloud` client, I didn't touch it anyhow manually, and I didn't use `docker login` by myself.

**Comment by artisticcheese** [comment]
Seeing the same issue here as well. Not using gcloud. Appears to be happening when remote registry is being authorized (I use AWS). Below is screenshow of what appears to be `Docker for Windows.exe` open and not closing file correctly.
![image](https://user-images.githubusercontent.com/13176884/28250165-1f010208-6a28-11e7-9341-658f5999bd5f.png)

**Comment by artisticcheese** [comment]
Solution seems to be add 2 second delay between loging to registry and any further commands
```
docker login -u $($tokenSegments[0]) -p $($tokenSegments[1]) $hostName
Start-Sleep 2
docker stack deploy --with-registry-auth --compose-file docker-stack-production.yml webredirect
```

**Comment by KingEmet** [comment]
Interesting. In my investigation, `Docker for Windows.exe` doesn't appear to ever relinquish the file.

**Comment by friism** [comment]
Thanks for reporting, I've opened an internal issue to track a fix.

**Comment by friism** [comment]
We've shipped fixes for this in both Stable (17.06.1) and Edge (17.07). Closing this issue, please comment or re-open if you're still seeing the problem after updating.
